app/src/androidTest/java/com/droidhats/campuscompass/views/MapFragmentTest.kt

```diff
@@ -54,12 +64,211 @@ class MapFragmentTest {
+        val navigateImageButton = onView(
+            allOf(
+                withId(R.id.startNavigationButton), withContentDescription("Navigate"),
+                isDisplayed()
+            )
+        )
+        navigateImageButton.check(matches(isDisplayed()))
+
+        //Checks if current location button is displayed
+        val currentLocationButton = onView(
+            allOf(
+                withId(R.id.myCurrentLocationFAB),
+                isDisplayed()
+            )
+        )
+        currentLocationButton.perform(click())
+
+        //Checks if "from" field is replaced by test "Your Location"
+        val currentLocationTextField = onView(
+            allOf(
+                withId(R.id.search_src_text), withText("Your Location"),
+                isDisplayed()
+            )
+        )
+        currentLocationTextField.check(matches(withText("Your Location")))
+
+        //Checks if Clear button for secondary (To) searchbar is displayed
+        val clearButton1ImageView = onView(
+            allOf(
+                withId(R.id.secondarySearchBar),
+                withChild(
+                    allOf(
+                        withId(R.id.search_bar),
+                        withChild(
+                            allOf(
+                                withId(R.id.search_edit_frame),
+                                withChild(
+                                    allOf(
+                                        withId(R.id.search_plate),
+                                        withChild(
+                                            withId(R.id.search_close_btn)
+                                        )
+                                    )
+                                )
+                            )
+                        )
+                    )
+                ),
+                isDisplayed()
+            )
+        )
+        clearButton1ImageView.check(matches(isDisplayed()))
+
+        //Checks if Clear button for main (from) searchbar is displayed
+        val clearButton2ImageView = onView(
+            allOf(
+                withId(R.id.mainSearchBar),
+                withChild(
+                    allOf(
+                        withId(R.id.search_bar),
+                        withChild(
+                            allOf(
+                                withId(R.id.search_edit_frame),
+                                withChild(
+                                    allOf(
+                                        withId(R.id.search_plate),
+                                        withChild(
+                                            withId(R.id.search_close_btn)
+                                        )
+                                    )
+                                )
+                            )
+                        )
+                    )
+                ),
+                isDisplayed()
+            )
+        )
+        clearButton2ImageView.check(matches(isDisplayed()))
```

Comment:
As mentioned earlier, avoid declaring variable for onView method specially since it is not being reused.